tests/protocol/test_wire.py

```diff
@@ -35,10 +35,10 @@
 from azurelinuxagent.common.version import CURRENT_VERSION, DISTRO_NAME, DISTRO_VERSION
 from tests.ga.test_monitor import random_generator
 from tests.protocol import mockwiredata
-from tests.protocol.mocks import mock_wire_protocol, mock_http_request
+from tests.protocol.mocks import mock_wire_protocol, HttpRequestPredicates as hrp
```

Comment:
Looks like a lot of changes, but it is just removing the nesting of mock_http_request inside of mock_wire_protocol, and using the utility functions in HttpRequestPredicates (plus fixing some tests that broke)